doc/src/base/base.md

```diff
@@ -435,6 +433,15 @@ Base.functionloc(::Method)
 Base.@locals
 ```
 
+## Code loading
+
```

Comment:
I added a new section as it felt a bit strange to keep these in "getting around"

Reply:
Correct me if I'm wrong. My current understanding is that executing these functions during the precompilation stage will be unstable, right?

For instance, the following code would almost certainly break someone's precompilation stage because `PkgExample` will be likely to be a `PkgId` or a `nothing` -- all depends on the current DEPOT state.

```julia
module Foo
    const PkgExample=identify_package("Example")

    ...
end
```

---

I asked this because my recent LazyModules.jl stuff uses `identify_package` to find out the UUID of a package [LazyModule(name)](https://github.com/johnnychen94/LazyModules.jl/blob/0031e2e88ba582d5d3fd0e0360c662df5de1d59c/src/LazyModules.jl#L16-L20) inside `@lazy import Example` macro. And this seems to be the cause of the encrypted precompilation error when I try to adopt the `@lazy` stuff to ImageIO https://github.com/JuliaRegistries/General/runs/6448677900?check_suite_focus=true

Reply:
Yes, currently we assume users will not use hacks to specify packages they have not correctly installed and registered to the current session. Perfectly acceptable to lazy-load it, it just must not attempt to work around the lookup logic. For example, `Base.identify_package` always requires using the 2 argument version when used inside packages.